.github/workflows/ci.yml

```diff
@@ -3,39 +3,51 @@ name: CI
 on: [push, pull_request]
 
 jobs:
-  lint:
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v2
-    - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v1
-      with:
-        node-version: 10.x
-    - name: Install Dependencies
-      run: npm install
-    - name: Lint
-      run: npm run lint
-
-  test:
-
+  tests:
     runs-on: ${{ matrix.os }}
 
+    timeout-minutes: 20
+
     strategy:
-      fail-fast: false
       matrix:
-        os: [macos-latest, ubuntu-latest, windows-latest]
-        node-version: [10.x]
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        node-version: [12.x, 14.x]
+
+    env:
+      HEADLESS: true
 
     steps:
+    - name: Prepare git
+      run: git config --global core.autocrlf false
+
     - uses: actions/checkout@v2
+
     - name: Use Node.js ${{ matrix.node-version }}
       uses: actions/setup-node@v1
       with:
         node-version: ${{ matrix.node-version }}
-    - name: Install Dependencies
+
+    - name: Get npm cache directory
+      id: npm-cache
+      run: echo "::set-output name=dir::$(npm config get cache)"
+
+    - uses: actions/cache@v2
+      with:
+        path: ${{ steps.npm-cache.outputs.dir }}
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: ${{ runner.os }}-node-
+
+    - name: Install Node.js dependencies
       run: npm install
-    - name: Tests
-      run: npm test
-      timeout-minutes: 10
-      env:
-        HEADLESS: true
+
+    - name: Code linter
+      run: npm run lint
+
+    - name: Browser tests
+      run: npm run test:browser
+
+    - name: Node.js tests
+      run: npm run test:node
+
+    - name: Code coverage
+      run: npm run posttest
```

Comment:
`npm run posttest` is to generate the test coverage report. Is this report accessible in CI? If it is not, it won't be very useful to run this step in CI.

Moreover, `posttest` reminds me of actions after running tests such as final clean up etc. I wonder if it makes sense to rename this npm task to a more meaningful name like "genTestCoverageRpt`?

Reply:
It will show up in the logs for the check. Later we should look at updating to use the codecov to report coverage. However, by separating this step out, in stead of just running `npm test` it's easier to find the results. Also the name `posttest` is related to the npm script so that it runs after the `test` script automatically. See: https://docs.npmjs.com/misc/scripts

Reply:
Great to know. Thanks for the explanation, @jobara.